### PR TITLE
Associations in action

### PR DESCRIPTION
**Pull request by dkukharau**
Hi all,

Here is associations example for contacts-app. Some new entities are added to the app, check them out!

### CODE REVIEW

**Comment by dkukharau** [comment]
I think this is a good time to think about how our API should look like for such a bit more complex cases, when there are several entities linked together. 
E.g. how adding/removing user to/from a group/profile should look like, should it be a separate method or done in scope of Create/Update.
If we're trying to build such end-to-end CRUD model, we need to think about what actually we want to support and to have as a default behavior. For now our API just reflects GORM behavior in terms of Create/Update.

File: pkg/pb/contacts.proto [hunk 1]
```diff
@@ -10,6 +10,169 @@ import "github.com/infobloxopen/protoc-gen-gorm/options/gorm.proto";
...
+
+message DeleteProfileRequest {
+    uint64 id = 1;
+}
+
+message ListProfilesResponse {
+    repeated Profile results = 1;
+}
+
+service Profiles {
```
[the remark this answers is not in the document]
Comment (dkukharau) [reply pkg/pb/contacts.proto]: It seems that neither proto nor golang support method overloading. Do you propose to change method naming to something like `(Create|Read|Update|Delete|List)ResourceName`? Sounds like an option, but the convention was different so far..

[the remark this answers is not in the document]
Comment (Calebjh) [reply pkg/pb/contacts.proto]: Without regard to which way is "better", protoc-gen-gorm should currently be agnostic to Create versus Create(TypeName) for method names.

[the remark this answers is not in the document]
Comment (dmacthedestroyer) [reply pkg/pb/contacts.proto]: I guess I should have been more specific... So long as protoc-gen-gorm can still handle both, I'm ok with either way.

**Pull request merged**